File: server/apps/sync-service/src/main/java/com/sharetimer/syncservice/config/WebClientConfig.java
package com.sharetimer.syncservice.config;

import java.util.Objects;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.client.WebClient;

@Configuration
public class WebClientConfig {

  @Bean("timerWebClient")
  public WebClient webClient(WebClientProps webClientProps) {
    return WebClient.builder().baseUrl(Objects.requireNonNull(webClientProps.getUrl()))
        .build();
  }

}

File: server/apps/sync-service/src/main/kotlin/com/sharetimer/syncservice/config/WebClientConfig.kt
package com.sharetimer.syncservice.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.client.WebClient

@Configuration
class WebClientConfig {
    @Bean("timerWebClient")
    fun webClient(webClientProps: WebClientProps): WebClient =
        WebClient.builder().baseUrl(requireNotNull(webClientProps.url)).build()
}
